File: designPatterns/Behavioral_patterns/Observer/python/Subject.py
from abc import ABC, abstractmethod


class Subject(ABC):
    """
    The Subject interface declares a set of methods for managing subscribers.
    """

    @abstractmethod
    def attach(self, observer):
        """
        Attach an observer to the subject.
        """
        pass

    @abstractmethod
    def detach(self, observer):
        """
        Detach an observer from the subject.
        """
        pass

    @abstractmethod
    def notify(self):
        """
        Notify all observers about an event.
        """
        pass

File: designPatterns/Behavioral_patterns/Observer/javascript/Subject.js
/**
 * The Subject interface declares a set of methods for managing subscribers.
 */
class Subject {
  // Attach an observer to the subject.
  attach(observer) {}

  // Detach an observer from the subject.
  detach(observer) {}

  // Notify all observers about an event.
  notify() {}
}

module.exports = Subject;
